### ISSUE TITLE
Error while deploying SKE cluster with terraform due to unknown values

### ISSUE THREAD

**Issue by docbyte86** [stackitcloud/terraform-provider-stackit#122]
Provider version 0.6.2

Error:

> │ Error: Value Conversion Error
> │ 
> │   with module.ske.stackit_ske_cluster.ske,
> │   on .terraform/modules/ske/main.tf line 12, in resource "stackit_ske_cluster" "ske":
> │   12:   extensions                  = var.ske_extensions
> │ 
> │ An unexpected error was encountered trying to build a value. This is always
> │ an error in the provider. Please report the following to the provider
> │ developer:
> │ 
> │ Received unknown value, however the target type cannot handle unknown
> │ values. Use the corresponding `types` package type or a custom type that
> │ handles unknown values.
> │ 
> │ Path: extensions
> │ Target Type: *ske.Extensions
> │ Suggested Type: basetypes.ObjectValue

**Child module:**

main.tf:

```
resource "stackit_ske_cluster" "ske" {
  project_id                  = var.ske_project_id
  name                        = var.ske_name
  kubernetes_version          = var.ske_version
  node_pools                  = var.ske_node_pool
  maintenance                 = var.ske_maintenance
  hibernations                = var.ske_hibernations
  extensions                  = var.ske_extensions
  allow_privileged_containers = var.ske_allow_privileged_containers
}
```

variables.tf

```
variable "ske_extensions" {
  description = "Specifies ACL and Argus Instance"
  type = object({
    acl = object({
      allowed_cidrs = list(string)
      enabled       = bool
    })
    argus = object({
      argus_instance_id = optional(string)
      enabled           = bool
    })
  })
}
```

**Root module:**

```
module "ske" {
  source         = "git::https://*****"
  ske_project_id = var.stackit_project_id
  ske_name       = local.ske_instance_name
  ske_version    = var.ske_version
  ske_node_pool  = var.ske_node_pool
  ske_extensions = {
    acl = {
      allowed_cidrs = ["0.0.0.0/0"]
      enabled       = true
    }
    argus = {
      argus_instance_id = "${module.argus.instance_id}"
      enabled           = true
    }
  }
}

```

I am not sure how to handle that error. IMHO I am not providing unknown values? I get this error during tf plan.

**Comment by docbyte86** [comment]
I think this could be related to
`      argus_instance_id = "${module.argus.instance_id}"`
In the stage of tf plan the id is unknown, because it will be created in the same main.tf

```
module "argus" {
  source         = "****"
  argus_name         = var.appcode
  stackit_project_id = var.stackit_project_id
  argus_plan         = "Monitoring-Starter-EU01"
}

module "ske" {
  source         = "****"
  ske_project_id = var.stackit_project_id
  ske_name       = local.ske_instance_name
  ske_version    = var.ske_version
  ske_node_pool  = var.ske_node_pool
  ske_extensions = {
    acl = {
      allowed_cidrs = ["0.0.0.0/0"]
      enabled       = true
    }
    argus = {
      enabled           = false
      argus_instance_id = "${module.argus.instance_id}"
    }
  }
}
```

**Comment by vicentepinto98** [comment]
Thank you for opening this issue.
This has been fixed in the v0.6.4 release.
